### PR TITLE
Fix nxos_hsrp and add unit test

### PR DESCRIPTION
**Pull request by trishnaguha**
##### SUMMARY

- Refactor nxos_hsrp
- Unit test

##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME

```modules/network/nxos/nxos_hsrp```

##### ANSIBLE VERSION

```
devel 2.4
```

### CODE REVIEW

File: lib/ansible/modules/network/nxos/nxos_hsrp.py
Comment (Qalthos) [inline lib/ansible/modules/network/nxos/nxos_hsrp.py]: Why can't we just return this? For that matter, what purpose does this method serve?

File: lib/ansible/modules/network/nxos/nxos_hsrp.py
Comment (Qalthos) [inline lib/ansible/modules/network/nxos/nxos_hsrp.py]: This might be an instance where we don't actually want this, I doubt `'None'` is a proper value here

**Review by Qalthos — CHANGES_REQUESTED** [review]

**Comment by mattclay** [comment]
CI failure in unit tests. Here's one of the failures:

```
>       return body
E       UnboundLocalError: local variable 'body' referenced before assignment

lib/ansible/modules/network/nxos/nxos_hsrp.py:141: UnboundLocalError
```

**Comment by trishnaguha** [comment]
@mattclay This is now fixed in https://github.com/ansible/ansible/commit/feafae70b579a3890951fd5c7c7b6645c340ff00

**Comment by trishnaguha** [comment]
Closing and reopening the PR to trigger shippable again.

File: lib/ansible/modules/network/nxos/nxos_hsrp.py
Comment (trishnaguha) [inline lib/ansible/modules/network/nxos/nxos_hsrp.py]: @Qalthos using lambda since the method would be one-liner in this case.

File: lib/ansible/modules/network/nxos/nxos_hsrp.py
Comment (trishnaguha) [inline lib/ansible/modules/network/nxos/nxos_hsrp.py]: Updated the change with lambda https://github.com/ansible/ansible/pull/24770/commits/f0c74e3a8f1bb368a8fdd3383a07d672ec04da37.

File: lib/ansible/modules/network/nxos/nxos_hsrp.py
Comment (trishnaguha) [reply lib/ansible/modules/network/nxos/nxos_hsrp.py]: Reverted the change https://github.com/ansible/ansible/pull/24770/commits/3bc4c09c91a5a17c7babaec670b10acfed8fc88a as it might return None https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/network/nxos/nxos_hsrp.py#L249-L258

File: lib/ansible/modules/network/nxos/nxos_hsrp.py
Comment (Qalthos) [inline lib/ansible/modules/network/nxos/nxos_hsrp.py]: So this is an astonishingly small nit, but we still don't need to get `value` from table, `for key in table:` will work fine here.

But that's probably not anything worth blocking the PR on

**Review by Qalthos — APPROVED** [review]

File: lib/ansible/modules/network/nxos/nxos_hsrp.py
Comment (trishnaguha) [reply lib/ansible/modules/network/nxos/nxos_hsrp.py]: Done :). Will merge it in after the CI tests pass.

**Pull request merged**
